text/3674-crates-io-remove-dev-deps.md

```diff
@@ -0,0 +1,43 @@
+- Start Date: 2024-07-31
+- RFC PR: [rust-lang/rfcs#3674](https://github.com/rust-lang/rfcs/pull/3674)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes to remove "dev-dependencies" from the git and sparse index of crates.io. 
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the index of crates.io contains information about dev-dependencies of crates. However, this information is [apparently not used by cargo](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Is.20dev-dependency.20information.20from.20the.20index.20used.3F) and never was. This unnecessary information increases the size of the index and thus the time it takes to download it. With the sparse index, the download time is already significantly reduced, but for crates with many dev-dependencies, the performance could still be improved.
+
+crates.io currently has about 9 million regular dependencies and 1.5 million dev-dependencies in its database. The dependencies part only contributes to a subset of the total size of the index, so it is unlikely that we will see a 15% size reduction in the index. Still, having 1.5 million dev-dependencies in the index is a significant amount of data that is not used by cargo and is unnecessarily transferred to the users of crates.io.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+From the user's perspective, this change should not have any impact. The `cargo` command will continue to work as before. The only difference is that the index will be smaller and the download time will be reduced.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The crates.io server will still process and save dev-dependencies in the database, but it will no longer include them in the index. To be more precise, any item in the `deps` field with `"kind": "normal"` will be removed from the index.
```

Comment:
I assume you meant

```suggestion
The crates.io server will still process and save dev-dependencies in the database, but it will no longer include them in the index. To be more precise, any item in the `deps` field with `"kind": "dev"` will be removed from the index.
```

Reply:
Had to read over that so many times because I kept thinking I was reading it wrong somehow.  Was still uncertain in posting it :)